src/auth/jsonfile_auth.rs

```diff
@@ -38,7 +36,7 @@ pub struct JsonFileAuthenticator {
 
 impl JsonFileAuthenticator {
     /// Initialize a new [`JsonFileAuthenticator`] from file.
-    pub fn new<T: Into<String>>(filename: T) -> Result<Self, Error> {
+    pub fn new<T: Into<String>>(filename: T) -> Result<Self, Box<dyn std::error::Error>> {
```

Comment:
One can probably use an enum of error types to avoid using dynamic dispatch here. Not a change I would make now but perhaps something to ponder about or experiment with.